src/DebugEngineHost/HostTelemetry.cs

```diff
@@ -50,6 +50,18 @@ public static void SendEvent(string eventName, params KeyValuePair<string, objec
         }
 
         /// <summary>
+        /// Initialize Host Telemetry
+        /// </summary>
+        [Conditional("LAB")]
+        public static void Initialize()
+        {
+            // Force the IVsTelemetryService to complete its lazy loading. There is a hang caused by trying to
+            // send telemetry while Visual Studio is launching the debugger if the telemetry helper also needs 
+            // to load the telemetry service on the main thread.
+            var telemetryService = TelemetryHelper.TelemetryService;
```

Comment:
You are going to need try/catches around this and probably to move this to another class. Otherwise you will break glass scenarios. See other uses of Telemetry for an example.